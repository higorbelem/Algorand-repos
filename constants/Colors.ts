const tintColorDark = '#fff';

export const Colors = {
  text: '#FFFFFF',
  background: '#191C21',
  tint: tintColorDark,
  icon: '#9BA1A6',
  tabIconDefault: '#9BA1A6',
  tabIconSelected: tintColorDark,
  main: '#2D2DF1',
  green: '#238636'
};
