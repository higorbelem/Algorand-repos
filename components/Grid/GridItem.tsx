import { StyleSheet, TouchableOpacity, View } from "react-native"
import { Image } from "expo-image";

import { Colors } from "@/constants/Colors";
import { githubRepos } from "@/@types/github";
import ForkIcon from "@/assets/svgs/fork.svg";
import IssueIcon from "@/assets/svgs/issue.svg";
import StarIcon from "@/assets/svgs/star.svg";
import { getLanguageColor } from "@/helpers/getLanguageColor";

import { Text } from "../Text";

export type GridItemProps = githubRepos & {
   onPress: () => void;
};

export const GridItem = (props: GridItemProps) => {

    return(
        <TouchableOpacity style={styles.container} activeOpacity={0.7} onPress={props.onPress}>
            <View style={styles.ownerContainer}>
                <Image source={{uri: props.owner.avatar_url}} style={styles.image}/>
                <Text size="small">{props.owner.login}</Text>
            </View>

            <Text weight="bold">{props.name}</Text>

            <Text size="small" numberOfLines={3} style={styles.description}>{props.description}</Text>

            <View style={styles.middleContainer}>
                <View style={styles.privateContainer}>
                    <Text size="small">{props.private ? 'Private' : 'Public'}</Text>
                </View>

                {
                    !!props.language && (
                        <View style={styles.languageContainer}>
                            <View style={[styles.languageDot, {backgroundColor: getLanguageColor(props.language)}]}/>
                            <Text size="small">{props.language}</Text>
                        </View>
                    )
                }
            </View>

            <View style={styles.statsContainer}>
                <View style={styles.statContainer}>
                    <StarIcon />
                    <Text size="small">{props.stargazers_count}</Text>
                </View>

                <View style={styles.statContainer}>
                    <ForkIcon />
                    <Text size="small">{props.forks}</Text>
                </View>

                <View style={styles.statContainer}>
                    <IssueIcon />
                    <Text size="small">{props.open_issues_count}</Text>
                </View>
            </View>
        </TouchableOpacity>
    )
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        borderColor: Colors.text,
        borderWidth: 1,
        borderRadius: 20,
        padding: 15,
        gap: 10
    },
    image: {
        width: 20,
        height: 20,
        borderRadius: 5
    },
    ownerContainer: {
        flexDirection: 'row',
        alignItems: 'center',
        gap: 5
    },
    middleContainer: {
        flexDirection: 'row',
        alignItems: 'center',
        gap: 10
    },
    privateContainer: {
        backgroundColor: '#ffffff22',
        paddingHorizontal: 10,
        paddingVertical: 5,
        borderRadius: 20
    },
    languageContainer: {
        flexDirection: 'row',
        alignItems: 'center',
        gap: 5
    },
    languageDot: {
        width: 10,
        height: 10,
        borderRadius: 5
    },
    statsContainer: {
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'space-between'
    },
    statContainer: {
        flexDirection: 'row',
        alignItems: 'center',
        gap: 5
    },
    description: {
        flex: 1
    }
});