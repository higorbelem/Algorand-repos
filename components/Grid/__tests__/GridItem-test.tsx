import * as React from 'react';
import { render } from '@testing-library/react-native';

import { reposMock } from '@/constants/mocks';

import { GridItem } from '../GridItem';

const mock = reposMock[0];

describe('GridItem component', () => {
  it('Renders correctly', () => {
    const tree = render(<GridItem {...mock} onPress={() => {}}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should show correct name', () => {
    const tree = render(<GridItem {...mock} onPress={() => {}}/>);

    const component = tree.getByText(mock.name);

    expect(component).toBeTruthy();
  });

  it('Should show correct description', () => {
    const tree = render(<GridItem {...mock} onPress={() => {}}/>);

    const component = tree.getByText(mock.description || '');

    expect(component).toBeTruthy();
  });

  it('Should show if private', () => {
    const tree = render(<GridItem {...mock} private={true} onPress={() => {}}/>);

    const component = tree.getByText('Private');

    expect(component).toBeTruthy();
  });

  it('Should show if public', () => {
    const tree = render(<GridItem {...mock} private={false} onPress={() => {}}/>);

    const component = tree.getByText('Public');

    expect(component).toBeTruthy();
  });

  it('Should show correct language', () => {
    const tree = render(<GridItem {...mock} onPress={() => {}}/>);

    const component = tree.getByText(mock.language || '');

    expect(component).toBeTruthy();
  });
});