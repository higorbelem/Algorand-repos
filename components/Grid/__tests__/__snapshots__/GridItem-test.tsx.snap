// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridItem component Renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderColor": "#FFFFFF",
      "borderRadius": 20,
      "borderWidth": 1,
      "flex": 1,
      "gap": 10,
      "maxWidth": 350,
      "opacity": 1,
      "padding": 15,
    }
  }
  testID="grid-item"
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 5,
      }
    }
  >
    <ViewManagerAdapter_ExpoImage
      borderRadius={5}
      containerViewRef={"[React.ref]"}
      contentFit="cover"
      contentPosition={
        {
          "left": "50%",
          "top": "50%",
        }
      }
      height={20}
      nativeViewRef={"[React.ref]"}
      onError={[Function]}
      onLoad={[Function]}
      onLoadStart={[Function]}
      onProgress={[Function]}
      placeholder={[]}
      source={
        [
          {
            "uri": "https://avatars.githubusercontent.com/u/98968122?v=4",
          },
        ]
      }
      style={
        {
          "borderRadius": 5,
          "height": 20,
          "width": 20,
        }
      }
      transition={null}
      width={20}
    />
    <Text
      size="small"
      style={
        [
          {
            "color": "#FFFFFF",
          },
          undefined,
          {
            "fontSize": 12,
            "fontWeight": "400",
          },
        ]
      }
    >
      perawallet
    </Text>
  </View>
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
        },
        undefined,
        {
          "fontSize": 16,
          "fontWeight": "700",
        },
      ]
    }
    weight="bold"
  >
    pera-wallet
  </Text>
  <Text
    numberOfLines={3}
    size="small"
    style={
      [
        {
          "color": "#FFFFFF",
        },
        {
          "flex": 1,
        },
        {
          "fontSize": 12,
          "fontWeight": "400",
        },
      ]
    }
  >
    Source code for Pera Wallet, simply the best Algorand wallet.
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 10,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#ffffff22",
          "borderRadius": 20,
          "paddingHorizontal": 10,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        size="small"
        style={
          [
            {
              "color": "#FFFFFF",
            },
            undefined,
            {
              "fontSize": 12,
              "fontWeight": "400",
            },
          ]
        }
      >
        Public
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 5,
        }
      }
    >
      <View
        style={
          [
            {
              "borderRadius": 5,
              "height": 10,
              "width": 10,
            },
            {
              "backgroundColor": "#F05138",
            },
          ]
        }
      />
      <Text
        size="small"
        style={
          [
            {
              "color": "#FFFFFF",
            },
            undefined,
            {
              "fontSize": 12,
              "fontWeight": "400",
            },
          ]
        }
      >
        Swift
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 5,
        }
      }
    >
      <svg />
      <Text
        size="small"
        style={
          [
            {
              "color": "#FFFFFF",
            },
            undefined,
            {
              "fontSize": 12,
              "fontWeight": "400",
            },
          ]
        }
      >
        190
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 5,
        }
      }
    >
      <svg />
      <Text
        size="small"
        style={
          [
            {
              "color": "#FFFFFF",
            },
            undefined,
            {
              "fontSize": 12,
              "fontWeight": "400",
            },
          ]
        }
      >
        60
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 5,
        }
      }
    >
      <svg />
      <Text
        size="small"
        style={
          [
            {
              "color": "#FFFFFF",
            },
            undefined,
            {
              "fontSize": 12,
              "fontWeight": "400",
            },
          ]
        }
      >
        22
      </Text>
    </View>
  </View>
</View>
`;
